plugins/woocommerce/src/Internal/Admin/WcPayWelcomePage.php

```diff
@@ -4,90 +4,57 @@
 
 use Automattic\WooCommerce\Admin\Features\OnboardingTasks\Tasks\WooCommercePayments;
 use Automattic\WooCommerce\Admin\WCAdminHelper;
+use Automattic\WooCommerce\Admin\PageController;
 
 /**
  * Class WCPayWelcomePage
  *
  * @package Automattic\WooCommerce\Admin\Features
  */
 class WcPayWelcomePage {
+	const TRANSIENT_NAME = 'wcpay_welcome_page_incentive';
+	const SCRIPT_NAME    = 'wcpay-welcome-page-incentive';
 
-	const EXPERIMENT_NAME = 'woocommerce_payments_menu_promo_us_2022';
-	const OTHER_GATEWAYS  = [
-		'affirm',
-		'afterpay',
-		'amazon_payments_advanced_express',
-		'amazon_payments_advanced',
-		'authorize_net_cim_credit_card',
-		'authorize_net_cim_echeck',
-		'bacs',
-		'bambora_credit_card',
-		'braintree_credit_card',
-		'braintree_paypal',
-		'chase_paymentech',
-		'cybersource_credit_card',
-		'elavon_converge_credit_card',
-		'elavon_converge_echeck',
-		'gocardless',
-		'intuit_payments_credit_card',
-		'intuit_payments_echeck',
-		'kco',
-		'klarna_payments',
-		'payfast',
-		'paypal',
-		'paytrace',
-		'ppcp-gateway',
-		'psigate',
-		'sagepaymentsusaapi',
-		'square_credit_card',
-		'stripe_alipay',
-		'stripe_multibanco',
-		'stripe',
-		'trustcommerce',
-		'usa_epay_credit_card',
-	];
+	/**
+	 * Plugin instance.
+	 *
+	 * @var WcPayWelcomePage
+	 */
+	protected static $instance = null;
+
+	/**
+	 * Main Instance.
+	 */
+	public static function instance() {
+		self::$instance = is_null( self::$instance ) ? new self() : self::$instance;
+
+		return self::$instance;
+	}
+	/**
+	 * Eligible incentive for the store.
+	 *
+	 * @var array|null
+	 */
+	private $incentive = null;
 
 	/**
 	 * WCPayWelcomePage constructor.
 	 */
 	public function __construct() {
-		add_action( 'admin_menu', array( $this, 'register_payments_welcome_page' ) );
+		add_action( 'admin_menu', [ $this, 'register_payments_welcome_page' ] );
+		add_filter( 'woocommerce_admin_shared_settings', array( $this, 'shared_settings' ) );
+		add_filter( 'woocommerce_admin_allowed_promo_notes', [ $this, 'allowed_promo_notes' ] );
 	}
 
 	/**
-	 * Registers the WooCommerce Payments welcome page.
+	 * Whether the WCPay welcome page is visible.
+	 *
+	 * @return boolean
 	 */
-	public function register_payments_welcome_page() {
-		global $menu;
-
-		// WC Payment must not be installed.
-		if ( WooCommercePayments::is_installed() ) {
-			return;
-		}
-
-		// Live store for at least 90 days.
-		if ( ! WCAdminHelper::is_wc_admin_active_for( DAY_IN_SECONDS * 90 ) ) {
-			return;
-		}
-
-		// Must be a US based business.
-		if ( WC()->countries->get_base_country() !== 'US' ) {
-			return;
-		}
-
-		// Has another payment gateway installed.
-		if ( ! $this->is_another_payment_gateway_installed() ) {
-			return;
-		}
-
-		// No existing WCPay account.
-		if ( $this->has_wcpay_account() ) {
-			return;
-		}
-
-		// Suggestions may be disabled via a setting.
+	public function must_be_visible() {
```

Comment:
Should we check if a user has already accepted an incentive for `welcome_page`? I imagine it would technically be possible that a user can accept multiple incentives otherwise.

Reply:
Actually, as the acceptance of any welcome page incentive will end with WCPay installed and an account created, it will not meet the conditions for the next incentive.

We also check on the server if by any chance the merchant try to apply a new welcome incentive while having another one 🙂